config/authsources.php

```diff
@@ -36,11 +36,10 @@
 
 $config = [];
 
-// Case for specifying no $SESSION IdP, select the first configured IdP as the default.
-$arr = array_reverse($saml2auth->metadataentities);
-$metadataentities = array_pop($arr);
-$idpentity = array_pop($metadataentities);
-$idp = md5($idpentity->entityid);
+// Grab the idp from the request if it's available
+if (isset($_GET['idp'])) {
+    $SESSION->saml2idp = $_GET['idp'];
```

Comment:
Is this supposed to be setting $SESSION->saml2idp  or $idp?

Reply:
You're right, it's meant to be $idp here, so it is set later in the $config variable